### DISCUSSION TITLE
Aspire.Hosting.Dcp.dcpctrl.ExecutableReconciler[0]: could not create Endpoint object

### DISCUSSION THREAD

**Discussion by Arthur-Lee** [microsoft/aspire#2803] [Q&A]
According to https://github.com/dotnet/aspire/issues/1499, I have succussed to compose local executable script to expose custom http endpoint, which can be injected into ASP.NET Core service. All the thing works as expected.

Currently I was prompted to update the library, which caused the strange error. And the ASP.NET Core can't access the local http service anymore.
![image](https://github.com/dotnet/aspire/assets/9337816/5e440259-f88e-4549-b03c-1789a621142a)

After some digging around, EndpointAnnotation in 8.0.0-preview.3.24105.21 doesn't contain isProxied filed, which maybe caused all the executable is proxied by DCP and caused the strange issue?

And I just found 8.0.0-preview.4.xxxx https://github.com/dotnet/aspire/blob/692dc41a65da572a7df25d53a9f2880afe59fdd8/src/Aspire.Hosting/ApplicationModel/EndpointAnnotation.cs#L1C1-L31C6 has added the filed.

Just curious is there any schedule to push the new previous version? and I just verify whether this issue is related to this filed.

**Comment by davidfowl** [comment]
What were you doing before?

**Comment by Arthur-Lee** [comment]
> What were you doing before?

Just normal orchestration which let ASP.NET Core web connect the backend ML GRPC service running in local windows. One day, I suddenly got an error that some namespace is not found. After update the Aspire library, the strange error happens without any code change.

**Comment by Arthur-Lee** [comment]
Following https://github.com/dotnet/aspire/issues/2050#issuecomment-1924495373 to upgrade aspire workload to 8.0.0-preview.4.24156.9/8.0.100 and set isProxied to false. It succeeds to resolve this issue. 

Another question, is there any easy way to share configuration between Aspire App Host and referenced Project? The class defined in ServiceDefaults project seems to not visible for Aspire App Host.
